packages/clarity-js/src/layout/region.ts

```diff
@@ -91,14 +97,16 @@ function handler(entries: IntersectionObserverEntry[]): void {
             // However, for larger regions, area of regions could be bigger than viewport and therefore comparison is relative to visible area
             let visible = viewportRatio > Setting.ViewportIntersectionRatio || entry.intersectionRatio > Setting.IntersectionRatio ? BooleanFlag.True : BooleanFlag.False;
             // If the visibility hasn't changed since the last tracked value, ignore this notification
-            if (!(id in regions && regions[id].visible === visible)) {
-                let d = { id, visible, region: regionMap.get(target) };
+            if (!(id in regions && visible && regions[id].state === Interaction.Visible)) {
```

Comment:
Minor suggestion is to move the state updating to a single function rather than having the logic spread into mulitple places like this. Something like

if(visibile) {
    updateState(id, Interaction.Visible)
}

Keeps the logic in one place and easier to track it.

Reply:
Good point & done